src/rail/estimation/algos/lephare.py

```diff
@@ -1,22 +1,155 @@
 
-class LephareEstimator(CatEstimator):
-    """LePhare-base CatEstimator
+
+class LephareInformer(CatInformer):
+    """Inform stage for LephareEstimator
+
+    This class will set templates and filters required for photoz estimation
     """
 
-    name = 'LephareEstimator'
+    name = "LephareInformer"
+    config_options = CatInformer.config_options.copy()
+    config_options.update(
+        zmin=SHARED_PARAMS,
+        zmax=SHARED_PARAMS,
+        nzbins=SHARED_PARAMS,
+        nondetect_val=SHARED_PARAMS,
+        mag_limits=SHARED_PARAMS,
+        bands=SHARED_PARAMS,
+        err_bands=SHARED_PARAMS,
+        ref_band=SHARED_PARAMS,
+        redshift_col=SHARED_PARAMS,
+    )
+
+    def __init__(self, args, comm=None):
+        """Init function, init config stuff (COPIED from rail_bpz)"""
+        CatInformer.__init__(self, args, comm=comm)
+        # Default local parameters
+        self.config_file = "lsst.para"
+        print("config file: " + self.config_file)
+        self.lephare_config = lp.read_config(self.config_file)
+
+    def _set_config(self, lephare_config):
+        """Update the lephare config
+
+        Parameters
+        ==========
+        lepahre_config : `dict`
+            A dictionary of the lephare config keywords.
+        """
+        self.lephare_config = lephare_config
+
+    def _create_filter_library(self):
+        """Make the filter library files in lephare format"""
+        # load filters from config file
+        filterLib = lp.FilterSvc.from_config(self.config_file)
+        # Get location to store filter files
+        filter_output = os.path.join(
+            os.environ["LEPHAREWORK"], "filt", self.lephare_config["FILTER_FILE"].value
+        )
+        # Write filter files
+        lp.write_output_filter(
+            filter_output + ".dat", filter_output + ".doc", filterLib
+        )
+
+    def _create_sed_library(self):
+        """Make the SED binary library files in lephare format.
+
+        We separately create the star, quasar and galaxy libraries.
+        """
+        sedlib = lp.Sedtolib(config_keymap=self.lephare_config)
+        sedlib.run(typ="STAR", star_sed="$LEPHAREDIR/examples/STAR_MOD_ALL.list")
+        sedlib.run(typ="QSO", qso_sed="$LEPHAREDIR/sed/QSO/SALVATO09/AGN_MOD.list")
+        sedlib.run(typ="GAL", gal_sed="$LEPHAREDIR/examples/COSMOS_MOD.list")
+
+    def _create_mag_library(self):
+        """Make the magnitudes library file in lephare format.
+
+        We separately create the star, quasar and galaxy libraries.
+
+        TODO: replace hardcoded config options with class config options.
+        """
+        maglib = lp.MagGal(config_keymap=self.lephare_config)
+        maglib.run(typ="STAR", lib_ascii="YES")
+        maglib.run(
+            typ="QSO",
+            lib_ascii="YES",
+            mod_extinc="0,1000",
+            eb_v="0.,0.1,0.2,0.3",
+            extinc_law="SB_calzetti.dat",
+        )
+        maglib.run(
+            typ="GAL",
```

Comment:
We were talking about this earlier, but yes, either these can be StageParam values that are passed in, or alternatively, we could pass this in as a single config dictionary StageParam.

It kind of depends on how this will be run. 
- Would it be typical to run three in a row like this, or would this ultimately just call `mag.lib` one time per stage? 
- Are there many variables beyond the ones that are listed? Will there be more? If yes to either, then perhaps dictionary makes more sense so that a change in input variables in LePhare doesn't require a change in StageParams in `rail_lephare`.